### PR TITLE
Enable support for AWS Credentials

### PR DESCRIPTION
**Pull request by hugomcfonseca**
https://plugins.jenkins.io/aws-credentials

### CODE REVIEW

**Comment by ctran** [comment]
Is there a way to do this (other secrets) without this change?

**Comment by hugomcfonseca** [comment]
@ctran I don't think so. The integration between this and the plugin `amazon-ecr` requires secrets' type to be `AWSCredentials`.

**Comment by jotadrilo** [comment]
Any update on this? Do you have plans to integrate it?

**Comment by Sliuza** [comment]
There is any update or plan to release this pr?

**Comment by jotadrilo** [comment]
@hugomcfonseca have you pinged the developers to check/revisit this?

**Comment by hugomcfonseca** [comment]
@jotadrilo I did not. I was expecting some reviews from them without pinging them. Going to ping them right away.

**Comment by hugomcfonseca** [comment]
@jtnord not sure who's the maintainers are, so I am pinging as you are one of the bigger contributors. Can I have your feedback on this PR? Thanks.

File: pom.xml [hunk 1]
```diff
@@ -151,11 +176,29 @@
...
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>structs</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins.workflow</groupId>
+      <artifactId>workflow-step-api</artifactId>
+    </dependency>
+    <dependency>
+        <groupId>com.fasterxml.jackson.module</groupId>
```
Comment (jtnord) [inline pom.xml]: why on earth do we need this dependency?  if we absolutely need it it should come from a plugin

File: src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java [hunk 2]
```diff
@@ -0,0 +1,270 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
```
Comment (jtnord) [inline src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java]: and 2019 you

File: src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertor.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
+/*
+ * The MIT License
+ *
+ * Copyright 2018 CloudBees, Inc.
```
Comment (jtnord) [inline src/main/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertor.java]: and 2019 you 😀

File: src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+            assertThat("credential description is mapped correctly", credential.getDescription(), is(emptyString()));
+            assertThat("credential scope is mapped correctly", credential.getScope(), is(CredentialsScope.GLOBAL));
+            assertThat("credential accessKey is mapped correctly", credential.getAccessKey(), is(accessKey));
+            assertThat("credential secretKey is mapped correctly", credential.getSecretKey().getPlainText(), is(secretKey));
+            assertThat("credential iamRoleArn is mapped correctly", credential.getIamRoleArn(), is(iamRoleArn));
+        }
+    }
+
+    @Test
+    public void failsToConvertWhenUsernameMissing() throws Exception {
```
Comment (jtnord) [inline src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java]: test name is incorrect, it is missing the access key

File: src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+            assertThat("credential description is mapped correctly", credential.getDescription(), is(emptyString()));
+            assertThat("credential scope is mapped correctly", credential.getScope(), is(CredentialsScope.GLOBAL));
+            assertThat("credential accessKey is mapped correctly", credential.getAccessKey(), is(accessKey));
+            assertThat("credential secretKey is mapped correctly", credential.getSecretKey().getPlainText(), is(secretKey));
+            assertThat("credential iamMfaSerialNumber is mapped correctly", credential.getIamMfaSerialNumber(), is(iamMfaSerialNumber));
+        }
+    }
+
+    @Test
+    public void canConvertAValidWithNoIamMfa() throws Exception {
```
Comment (jtnord) [inline src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java]: AValid*Secret*Wi....

File: src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java [hunk 6]
```diff
@@ -0,0 +1,270 @@
...
+            convertor.convert(secret);
+            fail("Exception should have been thrown");
+        } catch (CredentialsConvertionException cex) {
+            assertThat(cex.getMessage(), containsString("missing the accessKey"));
+        }
+    }
+
+
+    @Test
+    public void failsToConvertWhenPasswordMissing() throws Exception {
```
Comment (jtnord) [inline src/test/java/com/cloudbees/jenkins/plugins/kubernetes_credentials_provider/convertors/AWSCredentialsConvertorTest.java]: test name

**Comment by jtnord** [comment]
the pom changes adding new dependencies (except was creds)look very suspect to me, and a few method typos in the tests, otherwise looks ok.

sorry for the delay, for some reason I didn't see a notification about this

**Comment by hugomcfonseca** [comment]
@jtnord thanks, I am going to apply on your suggestions.

**Comment by hugomcfonseca** [comment]
@jtnord I finished applying your suggestions, please review

File: pom.xml [hunk 7]
```diff
@@ -151,11 +176,25 @@
       <groupId>org.jenkins-ci.plugins</groupId>
       <artifactId>credentials</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.jenkins-ci.plugins</groupId>
+      <artifactId>structs</artifactId>
+    </dependency>
```
[the remark this answers is not in the document]
Comment (hugomcfonseca) [reply pom.xml]: I put these dependencies in the wrong place. After adding support to AWS Credentials, I faced some issues regarding `RequireUpperBoundDeps`. I just follow this [guide](https://wiki.jenkins.io/display/JENKINS/How+to+fix+RequireUpperBoundDeps) to fix it and know it should be ok. 

Thanks again @jtnord.

**Comment by hugomcfonseca** [comment]
please rebuild

**Comment by hugomcfonseca** [comment]
@jtnord please review it again, I had to fix the conflicts on `pom.xml`

File: pom.xml [hunk 8]
```diff
@@ -197,14 +228,20 @@
...
-      
+
     <!-- static analysis -->
     <dependency>
       <groupId>com.github.spotbugs</groupId>
       <artifactId>spotbugs-annotations</artifactId>
       <scope>provided</scope>
       <optional>true</optional>
     </dependency>
+    <dependency>
```
Comment (jtnord) [inline pom.xml]: as this for fixing some upper-bounds issue?

Comment (hugomcfonseca) [reply pom.xml]: Yes @jtnord, the tests start to fail due to this:

```Caused by: java.io.IOException: Pipeline: Groovy v2.46 failed to load.
 - JavaScript GUI Lib: jQuery bundles (jQuery and jQuery UI) plugin v1.2 is older than required. To fix, install v1.2.1 or later.```

(if you want I can post the full stacktrace)

**Comment by jtnord** [comment]
thanks @hugomcfonseca

**Comment by hugomcfonseca** [comment]
Also thanks for the review, @jtnord. By the way, when should we expect a new version of the plugin?

**Comment by jtnord** [comment]
thanks @hugomcfonseca

**Comment by jtnord** [comment]
> when should we expect a new version of the plugin

sometime this week.

**Comment by Sliuza** [comment]
@jtnord ,the changes related to this PR was already released in the last version?

**Comment by jtnord** [comment]
nope, I got distracted. will do it tomorrow.

**Pull request merged**